File: 04910.py
import sys


def remainder_by_11(n):
    rem = 0
    for i, c in enumerate(n[::-1]):
        if i % 2 == 0:
            rem += int(c)
        else:
            rem -= int(c)
    return rem % 11


def divide_by_11(n):
    ret = "0"
    cur = 0
    for c in n:
        cur = cur * 10 + int(c)
        ret += str(cur // 11)
        cur %= 11
    return ret.lstrip("0")


def phone(n):
    r = remainder_by_11(n)
    if r == 0:
        return "IMPOSSIBLE"
    if len(n) % 2 == 0:
        m = str(11 - r) + n
    else:
        m = str(r) + n
    ret = divide_by_11(m)
    if len(ret) == len(n):
        return ret
    else:
        return "IMPOSSIBLE"


def main():
    test = 0
    while True:
        s = sys.stdin.readline().rstrip()
        if s == "0":
            break
        test += 1
        sys.stdout.write("{}. {}\n".format(test, phone(s)))


main()

File: 04910.cpp
#include <iostream>
#include <string>

using namespace std;

int remainder_by_11(string s){
	int ans = 0;
	for(unsigned i = 0; i < s.length(); ++i){
		if(i % 2 == 0){
			ans += s[i] - '0';
		}else{
			ans += 11 - (s[i] - '0');
		}
		ans %= 11;
	}
	if(s.length() % 2 == 0){
		ans = 11 - ans;
	}
	return ans % 11;
}

string divide_by_11(string s){
	string ret = "";
	int quot = 0;
	for(unsigned i = 0; i < s.length(); ++i){
		quot = quot * 10 + s[i] - '0';
		if(ret == "" && quot / 11 == 0){
			continue;
		}
		ret += (char)(quot / 11 + '0');
		quot %= 11;
	}
	return ret;
}

string phone(string s){
	int rem = remainder_by_11(s);
	if(rem == 0){
		return "IMPOSSIBLE";
	}
	if(s.length() % 2 == 0){
		rem = 11 - rem;
	}
	string ret = divide_by_11(to_string(rem) + s);
	// cout << ret << "\n";
	return ret.length() == s.length() ? ret : "IMPOSSIBLE";
}

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	string s;
	int t = 0;
	while(true){
		cin >> s;
		if(s == "0"){
			break;
		}
		++t;
		cout << t << ". " << phone(s) << "\n";
	}
	return 0;
}
